lms/static/scripts/frontend_apps/utils/format.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Truncate a URL to be `maxLength` or fewer characters.
+ *
+ * If `url` needs to be shortened, `truncateURL` tries to preserve the most
+ * informative parts of the URL using some simple heuristics.
+ *
+ * If `url` is not a valid URL, it is just shortened to the first `maxLength - 1`
+ * characters.
+ *
+ * @param {string} url
+ * @param {number} maxLength
+ */
+export function truncateURL(url, maxLength) {
```

Comment:
Any value in having a default `maxLength`? Maybe not.

Reply:
I'm not sure, so I've opted to require the caller to specify something appropriate for their context.